stock_serial_batch_receive/wizard/stock_move_line_serial_generator.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError, UserError
+
+
+class StockMoveLineSerialGenerator(models.TransientModel):
+
+    _name = 'stock.move.line.serial.generator'
+    _description = 'Auto generate serial numbers according to product sequence'
+
+    stock_move_id = fields.Many2one('stock.move', readonly=True)
+    product_id = fields.Many2one(
+        'product.product', related='stock_move_id.product_id'
+    )
+    picking_type_create_lots = fields.Boolean(
+        related='stock_move_id.picking_type_create_lots'
+    )
+    qty_to_process = fields.Integer(
+        help="Defines how many serial numbers will be generated on stock move "
+        "lines without a serial number yet."
+    )
+    first_number = fields.Char(copy=False)
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        stock_move_id = self.env.context.get('active_id')
+        move = self.env['stock.move'].browse(stock_move_id)
+        if 'qty_to_process' in fields:
+            lines_without_serial = move.move_line_ids.filtered(
+                lambda m: not m.lot_name
+            )
+            res['qty_to_process'] = len(lines_without_serial)
+        if 'first_number' in fields:
+            res['first_number'] = self._get_first_number(move)
+        return res
+
+    def _get_first_number(self, move):
```

Comment:
```suggestion
    @api.model
    def _default_first_number(self):
        """Default first number to use (can be inherited for custom behaviour)""""
        return '1'
```

Reply:
I think stock.move is right model to keep it

Reply:
But it has nothing to do with the move and is only used in the wizard to generate serials so why ?